products/views.py

```diff
@@ -1,3 +1,91 @@
-from django.shortcuts import render
+from django.http.response   import HttpResponseNotFound, JsonResponse
+from django.views           import View
+from django.db.models       import Q
 
-# Create your views here.
+from products.models import Category, Product
+
+class CategoryView(View):
+    def get(self, request):
+        categories = Category.objects.all()
+        result = [
+            {
+                "id"           : category.id,
+                "name"         : category.name,
+                "subcategories": [sub for sub in category.subcategory_set.values()]
+            }
+            for category in categories
+        ]
+        return JsonResponse({"result":result}, status=200)
+
+class ProductListView(View):
+    def get(self, request):
+        category_id    = request.GET.get("category_id")
+        subcategory_id = request.GET.get("subcategory_id")
+        offset         = request.GET.get("offset", 0)
+        limit          = request.GET.get("limit", 20)
+
+        q = Q()
+
+        if category_id:
+            q &= Q(category_id=category_id)
+        if subcategory_id:
+            q &= Q(subcategory_id=subcategory_id)
+
+        products = Product.objects.order_by('id').filter(q)[int(offset):int(offset)+int(limit)]
```

Comment:
위에서부터 int로 바꿔서 변수로 할당해준다면 실제 slice 부분 가독성이 더 좋아지겠네요.
```suggestion
        offset         = int(request.GET.get("offset", 0))
        limit          = int(request.GET.get("limit", 20))

        q = Q()

        if category_id:
            q &= Q(category_id=category_id)
        if subcategory_id:
            q &= Q(subcategory_id=subcategory_id)

        products = Product.objects.order_by('id').filter(q)[offset:offset+limit]
```